### PR TITLE
Issue 1282: base64_decode, base64_data - v8

### PR DESCRIPTION
**Pull request by jasonish**
Fixes issues in last PR: #1721
- Using existing string to integer conversion function.

Redmine ticket: https://redmine.openinfosecfoundation.org/issues/1282

Buildbot output:
- PR jasonish-pcap: https://buildbot.openinfosecfoundation.org/builders/jasonish-pcap/builds/156
- PR jasonish: https://buildbot.openinfosecfoundation.org/builders/jasonish/builds/157

### CODE REVIEW

File: src/detect-base64-decode.c [hunk 1]
```diff
@@ -0,0 +1,710 @@
...
+        goto error;
+    }
+    data->bytes = bytes;
+    data->offset = offset;
+    data->relative = relative;
+
+    if (s->list != DETECT_SM_LIST_NOTSET) {
+        sm_list = s->list;
+        if (data->relative) {
+            pm = SigMatchGetLastSMFromLists(s, 4,
```
Comment (inliniac) [inline src/detect-base64-decode.c]: <pre>
detect-base64-decode.c:214:13: warning: Value stored to 'pm' is never read
            pm = SigMatchGetLastSMFromLists(s, 4,
            ^    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
</pre>

File: src/detect-base64-decode.c [hunk 2]
```diff
@@ -0,0 +1,710 @@
...
+    if (!data->bytes) {
+        data->bytes = BASE64_DECODE_MAX;
+    }
+    if (data->bytes > de_ctx->base64_decode_max_len) {
+        de_ctx->base64_decode_max_len = data->bytes;
+    }
+
+    return 0;
+error:
+    return -1;
```
Comment (inliniac) [inline src/detect-base64-decode.c]: <pre>
detect-base64-decode.c:331:13: warning: Potential leak of memory pointed to by 'data'
    return -1;
            ^
</pre>

File: src/detect-base64-decode.c (hunk 1 above)
Comment (jasonish) [inline src/detect-base64-decode.c]: clang catch that?  I can't get that warning myself.  Its obviously correct, just trying to get the warning myself for future use.

Comment (jasonish) [inline src/detect-base64-decode.c]: Nevermind, got it. scan-build.

**Comment by jasonish** [comment]
See #1747.

**Pull request closed without merging**